cloudinit/util.py

```diff
@@ -68,6 +68,10 @@
                    ['lxc-is-container'])
 
 
+def kernel_version():
```

Comment:
please prepend this function with

```python
@lru_cache()
```

as the kernel version is unlikely to change during the runtime of cloud-init.